Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/survey/SurveyModule.kt

```diff
@@ -1,9 +1,62 @@
 package de.rki.coronawarnapp.datadonation.survey
 
+import android.content.Context
 import dagger.Module
+import dagger.Provides
+import dagger.Reusable
 import de.rki.coronawarnapp.datadonation.survey.consent.SurveyConsentModule
+import de.rki.coronawarnapp.datadonation.survey.server.DataDonationApiV1
+import de.rki.coronawarnapp.datadonation.survey.server.DataDonationHttpClient
+import de.rki.coronawarnapp.environment.datadonation.DataDonationCDNServerUrl
+import de.rki.coronawarnapp.http.HttpClientDefault
+import de.rki.coronawarnapp.http.RestrictedConnectionSpecs
+import de.rki.coronawarnapp.util.di.AppContext
+import okhttp3.Cache
+import okhttp3.ConnectionSpec
+import okhttp3.OkHttpClient
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import retrofit2.converter.protobuf.ProtoConverterFactory
+import java.io.File
+import javax.inject.Singleton
 
 @Module(
     includes = [SurveyConsentModule::class]
 )
-class SurveyModule
+class SurveyModule {
+
+    @Reusable
+    @DataDonationHttpClient
+    @Provides
+    fun cdnHttpClient(
+        @HttpClientDefault defaultHttpClient: OkHttpClient,
+        @RestrictedConnectionSpecs connectionSpecs: List<ConnectionSpec>
+    ): OkHttpClient =
+        defaultHttpClient.newBuilder().connectionSpecs(connectionSpecs).build()
+
+    @Singleton
+    @Provides
+    fun provideDataDonationApi(
+        @AppContext context: Context,
+        @DataDonationHttpClient client: OkHttpClient,
+        @DataDonationCDNServerUrl url: String,
+        protoConverterFactory: ProtoConverterFactory,
+        gsonConverterFactory: GsonConverterFactory
+    ): DataDonationApiV1 {
+        val cache = Cache(File(context.cacheDir, "http_data_donation"), DEFAULT_CACHE_SIZE)
+
+        val cachingClient = client.newBuilder().apply {
+            cache(cache)
+        }.build()
+
+        return Retrofit.Builder()
+            .client(cachingClient)
+            .baseUrl(url)
+            .addConverterFactory(protoConverterFactory)
+            .addConverterFactory(gsonConverterFactory)
+            .build()
+            .create(DataDonationApiV1::class.java)
```

Comment:
Would prefer `SurveyApiV1`.
DataDonation is the "parent" for all of this.
Survey and Analytics are the children and each one has their interface, we can maybe merge this later when all is done and approved, but as Analytics development is still on-going i'd prefer to keep naming and classes separate, below the level of the `@DataDonationCDNHttpClient`.

Reply:
renamed it as suggested